modules/core/src/main/java/org/apache/ignite/internal/processors/query/schema/SchemaIndexCachePartitionWorker.java

```diff
@@ -191,7 +192,10 @@ private void processPartition() throws IgniteCheckedException {
                         locked = false;
                     }
 
-                    cctx.cache().metrics0().addIndexRebuildKeyProcessed(1);
+                    GridCacheAdapter cache = cctx.cache();
```

Comment:
as i can see you already check cctx.cache() != null in SchemaIndexCachePartitionWorker#stop, why do we need additional check here ? + If we can obtain invariant that SchemaIndexCachePartitionWorker still in progress and GridCacheAdapter cache = cctx.cache(); cache **==** null we still can obtain flaky test, isn`t it ?